SPECS/ruby/ruby.spec

```diff
@@ -1,109 +1,105 @@
+%global erb_version             4.0.3
+%global error_highlight_version 0.6.0
+%global etc_version             1.4.3
+%global fcntl_version           1.1.0
+%global fiddle_version          1.1.2
+%global fileutils_version       1.7.2
+%global find_version            0.2.0
+%global forwardable_version     1.3.3
+%global getoptlong_version      0.2.1
+%global io_console_version      0.7.1
+%global io_nonblock_version     0.3.0
+%global io_wait_version         0.3.1
+%global ipaddr_version          1.2.6
+%global irb_version             1.13.1
+%global json_version            2.7.1
+%global logger_version          1.6.0
+%global mutex_m_version         0.2.0
+%global net_http_version        0.4.1
+%global net_protocol_version    0.2.2
+%global nkf_version             0.1.3
+%global observer_version        0.1.2
+%global open3_version           0.2.1
+%global openssl_version         3.2.0
+%global open_uri_version        0.4.1
+%global optparse_version        0.4.0
+%global ostruct_version         0.6.0
+%global pathname_version        0.3.0
+%global prism_version           0.19.0
+%global pp_version              0.5.0
+%global prettyprint_version     0.2.0
+%global pstore_version          0.1.3
+%global psych_version           5.1.2
+%global rdoc_version            6.6.3.1
+%global readline_version        0.0.4
+%global reline_version          0.5.7
+%global resolv_version          0.3.0
+%global resolv_replace_version  0.1.1
+%global rinda_version           0.2.0
 %global ruby2_keywords_version  0.0.5
-%global securerandom_version    0.2.0
-%global set_version             1.0.2
-%global shellwords_version      0.1.0
-%global singleton_version       0.1.1
-%global stringio_version        3.0.1.1
-%global strscan_version         3.0.1
-%global syslog_version          0.1.0
-%global tempfile_version        0.1.2
-%global time_version            0.2.2
-%global timeout_version         0.2.0
-%global tmpdir_version          0.1.2
-%global tsort_version           0.1.0
-%global un_version              0.2.0
-%global uri_version             0.12.2
-%global weakref_version         0.1.1
-%global win32ole_version        1.8.8
-%global yaml_version            0.2.0
-%global zlib_version            2.1.1
+%global securerandom_version    0.3.1
+%global set_version             1.1.0
+%global shellwords_version      0.2.0
+%global singleton_version       0.2.0
+%global stringio_version        3.1.1
+%global strscan_version         3.0.9
+%global syslog_version          0.1.2
+%global syntax_suggest_version  2.0.0
+%global tempfile_version        0.2.1
+%global time_version            0.3.0
+%global timeout_version         0.4.1
+%global tmpdir_version          0.2.0
+%global tsort_version           0.2.0
+%global un_version              0.3.0
+%global uri_version             0.13.1
+%global weakref_version         0.1.3
+%global win32ole_version        1.8.10
+%global yaml_version            0.3.0
+%global zlib_version            3.1.1
```

Comment:
The build failed. It looks like some of the modules listed in the long set of `Provides` later in the spec are now gone. If that's the case, please check what happened to them. If they were moved out of the standard Ruby libraries but not deprecated, then please re-add them as separate `rubygem-*` packages. See `rubygem-bigdecimal.spec` for example.

Reply:
@harshitgupta1337 kind ping?

Reply:
I am working on backporting the patch to the current version of ruby. This is because there are build failures when upgrading to 3.3.5.

Reply:
Please check this PR.
https://github.com/microsoft/azurelinux/pull/10506